File: BRaVeMobile/app/src/main/java/intl/iom/bravemobile/models/registrations/BiometricNotCollected.java
package intl.iom.bravemobile.models.registrations;

public class BiometricNotCollected {
    public int selectedReason = 1;
    public String reasonIfOther;

    public BiometricNotCollected(){
        selectedReason = 1;
    }
    public BiometricNotCollected(int selectedReason, String reasonIfOther)
    {
        this.selectedReason = selectedReason;
        this.reasonIfOther = reasonIfOther;
    }
}

File: BRaVe-Mobile-Backend/Models/BiometricNotCollected.cs
using System.Text.Json.Serialization;

namespace BRaVe_Mobile_Backend.Models
{
    public class BiometricNotCollected
    {
        [JsonPropertyName("selectedReason")]
        public int selectedReason { get; set; }

        [JsonPropertyName("reasonIfOther")]
        public string reasonIfOther { get; set; }

        public BiometricNotCollected() { }

        public BiometricNotCollected(int? selectedReason, string? reasonIfOther) { 

            this.selectedReason = selectedReason.HasValue? selectedReason.Value: 1;

            if (string.IsNullOrEmpty(reasonIfOther)) {
                this.reasonIfOther = "";
            }
            else
            {
                this.reasonIfOther = reasonIfOther;
            }     
        }
    }
}
